src/plugins/data_views/server/rest_api_routes/public/runtime_fields/create_runtime_field.ts

```diff
@@ -47,19 +47,23 @@ export const createRuntimeField = async ({
   runtimeField,
 }: CreateRuntimeFieldArgs) => {
   usageCollection?.incrementCounter({ counterName });
-  const dataView = await dataViewsService.get(id);
+  const dataView = await dataViewsService.getDataViewLazy(id);
 
-  if (dataView.fields.getByName(name) || dataView.getRuntimeField(name)) {
+  const fld = Object.keys((await dataView.getFields({ fieldName: [name] })).getFieldMap());
+
+  if (fld.length || dataView.getRuntimeField(name)) {
     throw new Error(`Field [name = ${name}] already exists.`);
   }
 
   const firstNameSegment = name.split('.')[0];
 
-  if (dataView.fields.getByName(firstNameSegment) || dataView.getRuntimeField(firstNameSegment)) {
+  const fld2 = Object.keys((await dataView.getFields({ fieldName: [name] })).getFieldMap());
```

Comment:
```suggestion
  const fld2 = Object.keys((await dataView.getFields({ fieldName: [firstNameSegment] })).getFieldMap());
```
Should we be using `firstNameSegment` here instead of `name` again?

Also, fairly minor, but I wonder if it would be better to fetch both fields at once instead of sending two separate requests.

Reply:
Good catch / optimization!